### PR TITLE
Simplify get key from extended methods

### LINKED ISSUE #295: Refactor HDWallet::getPublicKeyFromExtended
It has 6 parameters already, and if we want to move the logic from `Bitcon.swift` to C++, we need pass purpose or `DerivationPath` to C++.

### PR DESCRIPTION
**Pull request by alejandro-isaza**
Fix #295

### CODE REVIEW

File: swift/Tests/Blockchains/QtumTests.swift [hunk 1]
```diff
@@ -61,14 +61,4 @@ class QtumTests: XCTestCase {
         XCTAssertEqual(BitcoinAddress(publicKey: xpubAddr2, prefix: P2PKHPrefix.qtum.rawValue).description, "QStYeAAfiYKxsABzY9yugHDpm5bsynYPqc")
         XCTAssertEqual(BitcoinAddress(publicKey: xpubAddr9, prefix: P2PKHPrefix.qtum.rawValue).description, "QfbKFChfhx1s4VXS9BzaVJgyKw5a1hnFg4")
     }
-
-    func testDeriveFromZPub() {
```
Comment (hewigovens) [inline swift/Tests/Blockchains/QtumTests.swift]: I think we should keep bip84/zpub tests?

File: tests/interface/TWGroestlcoinTests.cpp [hunk 2]
```diff
@@ -78,8 +78,8 @@ TEST(Groestlcoin, ExtendedKeys) {
 
 TEST(Groestlcoin, DeriveFromZpub) {
     auto zpub = STRING("zpub6qXFnWiY6FdT5BQptrzEhHfm1WpaBTFc6MHzR4KwscXGdt6xCqUtrAEjrHdeEsjaYEwVMgjtTvENQ83yo2fmkYYGjTpJoH7vFWKQJp1bg1X");
-    auto pubKey4 = TWHDWalletGetPublicKeyFromExtended(zpub.get(), TWCoinTypeGroestlcoin, TWHDVersionZPUB, TWHDVersionZPRV, 0, 4);
-    auto pubKey11 = TWHDWalletGetPublicKeyFromExtended(zpub.get(), TWCoinTypeGroestlcoin, TWHDVersionZPUB, TWHDVersionZPRV, 0, 11);
+    auto pubKey4 = TWHDWalletGetPublicKeyFromExtended(zpub.get(), STRING("m/44'/17'/0'/0/4").get());
```
Comment (hewigovens) [inline tests/interface/TWGroestlcoinTests.cpp]:
```suggestion
    auto pubKey4 = TWHDWalletGetPublicKeyFromExtended(zpub.get(), STRING("m/84'/17'/0'/0/4").get());
```

File: tests/interface/TWGroestlcoinTests.cpp [hunk 3]
```diff
@@ -78,8 +78,8 @@ TEST(Groestlcoin, ExtendedKeys) {
...
+    auto pubKey11 = TWHDWalletGetPublicKeyFromExtended(zpub.get(), STRING("m/44'/17'/0'/0/11").get());
```
Comment (hewigovens) [inline tests/interface/TWGroestlcoinTests.cpp]:
```suggestion
    auto pubKey11 = TWHDWalletGetPublicKeyFromExtended(zpub.get(), STRING("m/84'/17'/0'/0/11").get());
```

File: tests/interface/TWHDWalletTests.cpp [hunk 4]
```diff
@@ -194,31 +194,10 @@ TEST(HDWallet, PublicKeyFromX) {
...
-    auto ypubAddr10 = TWHDWalletGetPublicKeyFromExtended(ypub.get(), TWCoinTypeBitcoin, TWHDVersionYPUB, TWHDVersionYPRV, 0, 10);
-
-    auto data3 = WRAPD(TWPublicKeyData(ypubAddr3));
-    auto data10 = WRAPD(TWPublicKeyData(ypubAddr10));
-
-    assertHexEqual(data3, "0299bd0bdc081a9888fac95a33e8bebcdeeb57cf7477f2f0721362f3a51a157227");
-    assertHexEqual(data10, "03a39ad9c0d19bb43c45643582614298c96b0f7c9462c0de789c69013b0d609d1c");
-}
-
 TEST(HDWallet, PublicKeyFromZ) {
```
Comment (hewigovens) [inline tests/interface/TWHDWalletTests.cpp]: use m/84' for zpub

File: tests/interface/TWViacoinTests.cpp [hunk 5]
```diff
@@ -74,27 +74,10 @@ TEST(Viacoin, ExtendedKeys) {
     assertStringsEqual(zpub, "zpub6qUUfiw2RW2i3Sy8zHVgbjoQSsi4UkgC75e6SjorAFwEV7YGFQoLELRhpe8NZ3xtsKHhbJW3kYWQLMFfCmMFDPkSer8BP9fiJcaZRL4LsR4");
 }
 
-TEST(Viacoin, DeriveFromXpub) {
```
Comment (hewigovens) [inline tests/interface/TWViacoinTests.cpp]: like Qtum, let's keep xpub/zpub tests

File: swift/Tests/Blockchains/QtumTests.swift (hunk 1 above)
Comment (alejandro-isaza) [reply swift/Tests/Blockchains/QtumTests.swift]: In the current state they fail. Qtum defines its HDVersion as XPUB

**Pull request merged**